application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -57,8 +57,8 @@
       ## Chck if the current user has delete right on the forum
       #if($webHomeDoc.hasAccessLevel('delete'))
         ## We need to delete all documents of the forum space
-        #set($result = $services.query.hql("where doc.space= :forum").bindValue('forum', $space).execute())
-        #foreach($fullName in $result)
+        #set($results = $services.query.xwql('where doc.space like :forum').bindValue('forum', "${space}%").execute())
```

Comment:
Missing dot from `"${space}%"`. Should be `"${space}%."`, otherwise space `A` would be "LIKE" space `AB`.

You also need to include the forum document itself, so you also need a second clause but which checks for equality between doc.space and the passed $space value.

What you want to get are all documents that are in the actual $space OR in a subspace of $space... so:
```
WHERE doc.space=:forumSpace OR doc.space LIKE :forum
```
...binding `forumSpace` to `$space` and `forum` to `"${space}%."`.